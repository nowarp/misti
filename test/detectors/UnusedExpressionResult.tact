fun fun1() {}
fun fun2(): Int { return 42; }
fun test() {
    fun1(); // OK
    fun2(); // Bad
    1; // Bad
    1 == 1; // Bad

    let a = 42;
    a == 43; // Bad

    require(1 == 1, "1"); // OK: This function is not defined in AST

    send(SendParameters{ to: sender(), value: 42 }); // OK: Ignored function
}

// Check contract methods
contract C1 {
    fun method1() {} // to check method name collision with C2.method1
    fun test() {
        self.method1(); // OK: Calling method returning void
    }
}
contract C2 {
    fun method1(): Int { return 42; }
    fun test() {
        self.method1(); // Bad: Calling method returning Int
    }
}

// Check methods defined in parent traits
trait T1 {
    fun traitMethod1(): Int { return 42; }
    fun traitMethod2() {}
}
trait T2 with T1 { fun traitMethod3(): Int { return 42; } }
contract CT with T2 {
    fun test() {
        self.traitMethod1(); // Bad
        self.traitMethod2(); // OK
        self.traitMethod3(); // Bad
    }
}
