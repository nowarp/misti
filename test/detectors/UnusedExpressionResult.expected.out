[MEDIUM] UnusedExpressionResult: Result of evaluation of fun2() is unused
test/detectors/UnusedExpressionResult.tact:5:5:
  4 |     fun1(); // OK
> 5 |     fun2(); // Bad
          ^
  6 |     1; // Bad
Help: Remove the expression or assign its result
See: https://nowarp.io/tools/misti/docs/detectors/UnusedExpressionResult

[MEDIUM] UnusedExpressionResult: Result of evaluation of 1 is unused
test/detectors/UnusedExpressionResult.tact:6:5:
  5 |     fun2(); // Bad
> 6 |     1; // Bad
          ^
  7 |     1 == 1; // Bad
Help: Remove the expression or assign its result
See: https://nowarp.io/tools/misti/docs/detectors/UnusedExpressionResult

[MEDIUM] UnusedExpressionResult: Result of evaluation of 1 == 1 is unused
test/detectors/UnusedExpressionResult.tact:7:5:
  6 |     1; // Bad
> 7 |     1 == 1; // Bad
          ^
  8 | 
Help: Remove the expression or assign its result
See: https://nowarp.io/tools/misti/docs/detectors/UnusedExpressionResult

[MEDIUM] UnusedExpressionResult: Result of evaluation of a == 43 is unused
test/detectors/UnusedExpressionResult.tact:10:5:
   9 |     let a = 42;
> 10 |     a == 43; // Bad
           ^
  11 | 
Help: Remove the expression or assign its result
See: https://nowarp.io/tools/misti/docs/detectors/UnusedExpressionResult

[MEDIUM] UnusedExpressionResult: Result of evaluation of self.method1() is unused
test/detectors/UnusedExpressionResult.tact:27:9:
  26 |     fun test() {
> 27 |         self.method1(); // Bad: Calling method returning Int
               ^
  28 |     }
Help: Remove the expression or assign its result
See: https://nowarp.io/tools/misti/docs/detectors/UnusedExpressionResult

[MEDIUM] UnusedExpressionResult: Result of evaluation of self.traitMethod1() is unused
test/detectors/UnusedExpressionResult.tact:39:9:
  38 |     fun test() {
> 39 |         self.traitMethod1(); // Bad
               ^
  40 |         self.traitMethod2(); // OK
Help: Remove the expression or assign its result
See: https://nowarp.io/tools/misti/docs/detectors/UnusedExpressionResult

[MEDIUM] UnusedExpressionResult: Result of evaluation of self.traitMethod3() is unused
test/detectors/UnusedExpressionResult.tact:41:9:
  40 |         self.traitMethod2(); // OK
> 41 |         self.traitMethod3(); // Bad
               ^
  42 |     }
Help: Remove the expression or assign its result
See: https://nowarp.io/tools/misti/docs/detectors/UnusedExpressionResult